### ISSUE TITLE
Package request: nginx module `nginx-otel`

### ISSUE THREAD

**Issue by spacekitteh** [NixOS/nixpkgs#355118]
**Project description**
Adds opentelemetry capabilities to nginx. I have tried packaging it myself, but I can't figure out how to get it to use the nginx sources.

```nix
  otel = {
    name = "otel";
    inputs = with pkgs; [cmake];
    src = pkgs.fetchFromGitHub {
      name = "otel";
      owner = "nginxinc";
      repo = "nginx-otel";

      #2024-11-11
      rev = "1074d02758156a8c2e5a9e32f16293353226e830";
      hash = "sha256-HZc+nizyHFFal7WM3DKdCJ/HD/inh22/WPEHro/vzcI=";
    };
    meta = with lib; {
      description = "NGINX OpenTelemetry instrumentation module";
      homepage = "https://github.com/nginxinc/nginx-otel";
      license = with licenses; [asl20];
      maintainers = [];
    };
  };
```

**Metadata**

* homepage URL: https://github.com/nginxinc/nginx-otel
* source URL: https://github.com/nginxinc/nginx-otel
* license: asl20
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by ibizaman** [comment]
FWIW adding the definition above to `services.nginx.additionalModules` results for me in the following error:

```
CMake Error: The source directory "/build/nginx-1.26.2" does not appear to contain CMakeLists.txt.
```

**Comment by ibizaman** [comment]
Yes, clearly this is not really possible currently https://github.com/NixOS/nixpkgs/issues/258260#issuecomment-1759951513

**Comment by auscyber** [comment]
this needs to be reopened given https://github.com/NixOS/nixpkgs/issues/258260
